pages/admin/index.tsx

```diff
@@ -266,13 +349,32 @@ export default function Requests() {
                   <InputLeftElement pointerEvents="none">
                     <SearchIcon color="text.filler" />
                   </InputLeftElement>
-                  <Input placeholder="Search by first name, last name or user ID" />
+                  <Input
+                    placeholder="Search by name or user ID"
+                    onKeyDown={event => {
+                      if (event.key === 'Enter') {
+                        setSearchFilter(searchInput);
+                        setPageNumber(0);
+                      }
+                    }}
+                    onChange={event => setSearchInput(event.target.value)}
+                  />
                 </InputGroup>
               </Box>
             </Flex>
-            <Table columns={COLUMNS} data={DATA} />
+            {!loading && networkStatus !== NetworkStatus.refetch && (
+              <Table
+                columns={COLUMNS}
+                data={requestsData || []}
+                onChangeSortOrder={sortOrder => setSortOrder(sortOrder)}
+              />
+            )}
             <Flex justifyContent="flex-end">
-              <Pagination pageSize={20} totalCount={150} />
+              <Pagination
+                pageSize={PAGE_SIZE}
+                totalCount={recordsCount}
+                onPageChange={n => setPageNumber(n)}
+              />
```

Comment:
also fixed a bug in the pagination component in my latest commit. It now sets the page number in the props instead of keeping the logic inside the component which allows the pageNumber to be set to 0 after a filter change